### DISCUSSION TITLE
Error while installing AdobeAIR.exe on client pc.

### DISCUSSION THREAD

**Discussion by balaedit** [airsdk/Adobe-Runtime-Support#2788] [General]
While installing adobe air runtime on client pc getting error that this pc required a version of Adobe AIR which cannot be found.

image of error and pc configuration below : 
![harmanAir](https://github.com/airsdk/Adobe-Runtime-Support/assets/37571867/0b8fb088-27d9-462f-85e9-d4d4919c02b8)
![pcconfig](https://github.com/airsdk/Adobe-Runtime-Support/assets/37571867/c1c328bf-c10c-405e-84c0-85685860fa33)

@ajwfrost Please share that how to install Adobe AIR if this issue occurred while installing Adobe AIR not even the app that I have build. 

Thanks.

**Comment by ajwfrost** [comment]
Hi

It seems something is preventing the runtime from being opened .. which version is this?

Could you try opening the "Adobe AIR.exe" file using a zip file manager, and unzip it somewhere? Inside is the "Adobe AIR\Versions\1.0" folder with "Adobe AIR.dll" in there.. would it be possible for you to download the Windows dependency walker (https://dependencywalker.com/) and use this to open the DLL file - it can take quite a lot of time to open. But then it should be possible to check the top-level dependencies of AIR and see if there are any missing imports from any of them..

thanks

**Reply by balaedit** [reply]
I had downloaded the Adobe AIR from https://airsdk.harman.com/runtime

as you told I unzipped the Adobe AIR.exe and open the Adobe AIR.dll in the dependencyWalker application, it display message as below: 

![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/37571867/7c21f040-f4eb-44af-b1f5-d63b1b42ccb2)

**Reply by ajwfrost** [reply]
Thanks - yes there are always issues reported, but if you do View -> "Collapse all" which collapses all the items from the tree-view in the top left, and then click on the + next to Adobe AIR.dll and you'll get the top level of items..
Then if you click on each of those in turn, it should show a list of the imports that are expected from that library in the top-right pane.. if any of those are red (they'd be at the top of the list) it means we have a dependency issue.....

![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/6400804/f2e9483e-372c-435a-bb54-e4cefff64323)

**Reply by balaedit** [reply]
Thanks

I got your point. What should I do to install Adobe air in client pc. Or should I send the details of what I get error from import from dll. Or you will send the updated Adobe air.

**Reply by ajwfrost** [reply]
We still need to find out whether there were any missing imports, unless I missed an email? So if you can get your Dependency Walker so that it looks roughly like my screenshot, and then you click on each of the left-hand top level dependencies one after another, do you see any of the entries in the top right box appear to have red boxes (vs the green boxes you can see there)?

Most likely thing is an API that we're compiling against that wasn't present in WIndows 7..  but - I believe we still have a Windows 7 machine lying around in the office that we can also try this out on ourselves, next week hopefully.

thanks

**Reply by balaedit** [reply]
I got the details of the missing module list in the Adobe AIR.dll in the below text file :
using depends22_x64 -
[DLLerrors.txt](https://github.com/airsdk/Adobe-Runtime-Support/files/12443893/DLLerrors.txt)
using depends22_x86
[DLLerrors_x86.txt](https://github.com/airsdk/Adobe-Runtime-Support/files/12443909/DLLerrors_x86.txt)

Don't know whether it will be helpful. Sorry but I want to get this issue resolved as soon as possible.

**Reply by ajwfrost** [reply]
Hi - sorry, those text files aren't going to help, the tool comes up with a lot of outputs which aren't actually valid/relevant. The key thing is to see whether any of the direct dependencies have got imports that are missing..

One other thing to check: have you tried running the "Adobe AIR Installer.exe" file, having unzipped that initial exe file? I wasn't sure if this would give a more useful error message. Also, can you check if there's anything being output to the log file:
C:\Users\username\AppData\Local\Adobe\AIR\logs\install.log

thanks

**Comment by balaedit** [comment]
> We still need to find out whether there were any missing imports, unless I missed an email? So if you can get your Dependency Walker so that it looks roughly like my screenshot, and then you click on each of the left-hand top level dependencies one after another, do you see any of the entries in the top right box appear to have red boxes (vs the green boxes you can see there)?
> 
> Most likely thing is an API that we're compiling against that wasn't present in WIndows 7.. but - I believe we still have a Windows 7 machine lying around in the office that we can also try this out on ourselves, next week hopefully.
> 
> thanks

Is there any update on this issue for Windows 7 / Vista?

**Reply by ajwfrost** [reply]
Hi

We've just checked this and it installs fine on our Windows 7 machine. Below is a screenshot of the dependency walker screen for it.. can you check that you have all of the DLL files that it lists on the left hand pane there? All but one are expected to be in c:\windows\system32 .. the path for this "comctl32.dll" that's highlighted may be different on your machine but I would expect it to be listed .. but worth checking if you highlight it, that the ImageList functions mentioned in the top right are shown properly with green boxes.

Failing that, it would just be a case if you can use Dependency Walker to get to a similar view, just select each of those top-level dependency DLLs in turn, and make sure the top right pane doesn't have any entries that start with a red rectangle....


thanks

![air_dependencies](https://github.com/airsdk/Adobe-Runtime-Support/assets/6400804/83f22740-1263-4e4d-a83e-710d56b3afdd)
